### PR TITLE
Unit DB Feature parity PR

### PR DESCRIPTION
**Pull request by Raistlfiren**
This is currently a placeholder PR as I work on updating code to fix weapons, add DoTPulses, fix fire cycle, add more weapon details, add link to Github blueprint, and add stat highlighting.

This should knock out the following issues - https://github.com/spooky/unitdb/issues/56, https://github.com/spooky/unitdb/issues/54, https://github.com/spooky/unitdb/issues/53

Current todo list:
- [x] Fix fire cycles
- [x] Add DoTPulses
- [x] Add Firing Randomness
- [x] Add Firing Lifetime
- [x] Fix T3 aeon rapid fire artillery
- [x] Fix beam fire cycle
- [x] Add beam cycle
- [x] Link unit to Github blueprint
- [x] Fix miscellaneous weapon issues

### CODE REVIEW

**Comment by Raistlfiren** [comment]
@spooky - Please review the PR when you get a chance. I have fixed all of the problems with weapons now, and I have added more information about beams. I also added a link from the blueprint ID to the specific Github page for easy reference. (This might need to be more obvious than what it currently is.)

As an aside, I would like to link/include this unit database as the main database now on the new website. Do you have any problems with connecting this repository to the official Faforever organization? Or at least me adding a `Dockerfile` so it can be deployed on the Faforever servers?

**Comment by spooky** [comment]
@Raistlfiren thanks for the PR, will need a bit more time to review/merge though. As for connecting this to faforever & linking/installing/sharing - no problems with that, this is open source and is meant to serve the community.

**Comment by Raistlfiren** [comment]
@spooky - thank you for taking the time to review the PR. I will try to discuss the merge with downlord and sheeo.

**Comment by Raistlfiren** [comment]
@spooky - After speaking with downlord on slack, he said that we can transfer the repository to the organization and you would still have write access to the repository at first. Admin access would need to be explicitly given to you by visionik or sheeo. I will try to reach out to them as well.

**Comment by Raistlfiren** [comment]
@spooky - I just wanted to check in and see if you have been able to review it and think about moving to the Faforever organization. Downlord should be able to help you with the transition. If you would like to talk further, then we can discuss it on slack.

**Comment by spooky** [comment]
@Raistlfiren will do, just can't seem to find the time. When I do, I'll catch someone on slack and try to push this further.

File: app/js/UnitDecorator.js [hunk 1]
```diff
@@ -123,20 +123,118 @@ unitDb.UnitDecorator = function(blueprint) {
...
         getDps = function(weapon) {
-            return unitDb.dpsCalculator.dps(weapon);
+            var stats = weaponStats(weapon);
+
+            if (weapon.ForceSingleFire)
+                return null;
+
+            return unitDb.dpsCalculator.dps(weapon, stats);
+        },
+        isTML = function(weapon) {
```
Comment (spooky) [inline app/js/UnitDecorator.js]: isTML = function(weapon) { return !!weapon.ForceSingleFire; }

File: app/js/UnitDecorator.js [hunk 2]
```diff
@@ -123,20 +123,118 @@ unitDb.UnitDecorator = function(blueprint) {
...
+            if (weapon.ForceSingleFire)
+                return true;
+
+            return false;
```
Comment (spooky) [inline app/js/UnitDecorator.js]: there's a lot of dense code in this file - this should be extracted to smaller methods - right now it's really hard to follow.

File: test/spec/dps.js [hunk 3]
```diff
@@ -27,7 +27,8 @@ describe('dps', function () {
         'TurretYawSpeed': 100,
         'WeaponCategory': 'Direct Fire'
     };
-    var dps = unitDb.dpsCalculator.dps(weapon);
+    var stats = JSON.parse('{"shots":1,"cycle":0.3,"damage":8,"salvoDelay":0,"salvoSize":1,"dotPulse":1}');
```
Comment (spooky) [inline test/spec/dps.js]: JSON.parse is really not needed here... this is js, you can just create the object directly. That goes for all the JSON.parse calls that have been added in this file.

File: tools/cleaner.py [hunk 4]
```diff
@@ -398,12 +398,12 @@ def run(app_path):
     unit_data_path = os.path.join(app_path, 'data', 'index.json')
     unit_dataFat_path = os.path.join(app_path, 'data', 'index.fat.json')
 
-    unit_list = load_unit_file(unit_data_path)
+    unit_list = load_unit_file(unit_dataFat_path)
```
Comment (Raistlfiren) [inline tools/cleaner.py]: Well, the problem is that the method `slenderize` is being called AFTER the file is loaded. I was running into issues where some of the new properties like `FiringRandomness`, `FiringTolerance`, and etc... wasn't being added to my `index.json` file because the `index.json` file only included the slimmed down version of properties. Unless I am still not clear about this, please verify. Thanks!

File: test/spec/dps.js (hunk 3 above)
Comment (Raistlfiren) [inline test/spec/dps.js]: This is silly. I don't really know why I did this. Thanks for the catch and I will update my PR soon with these changes.

File: tools/cleaner.py (hunk 4 above)
Comment (spooky) [inline tools/cleaner.py]: Slenderize is ment to throw out all the properties that are not used so that the index is smaller in size. The issue you need to solve is why slenderize doesn't find the new properties.

Comment (spooky) [inline tools/cleaner.py]: This may help:
https://github.com/spooky/unitdb/blob/master/tools/cleaner.py#L349

Comment (Raistlfiren) [inline tools/cleaner.py]: Yeah, exactly. The problem was that `index.json` did not include the methods and so they were not being included in the updated output. Once I changed that to the `fat` JSON file, then I was able to get all of the new methods and include them in the outputted file - `index.json`. The methods weren't included because they were never included initially.

**Comment by spooky** [comment]
merged changes in 931f00bf2018145a05288f450909f94494d0fbdc
